code/game/objects/items/toys/toys.dm

```diff
@@ -538,6 +542,217 @@
+	var/randomchoice = rand(1,8)
+	switch(randomchoice)
+		if(1)
+			pick(playsound(src, 'sound/items/gnome.ogg', 35, TRUE),
+			playsound(src, 'sound/misc/robotic scream.ogg', 35, TRUE),
+			playsound(src, 'sound/voice/pred_laugh1.ogg', 35, TRUE),
+			playsound(src, 'sound/voice/pred_laugh2.ogg', 35, TRUE),
+			playsound(src, 'sound/voice/pred_laugh3.ogg', 35, TRUE),
+			playsound(src, 'sound/voice/gnomelaugh.ogg', 35, TRUE),
+			playsound(src, 'sound/weapons/guns/fire/tank_cannon1.ogg', 35, TRUE),
+			playsound(src, 'sound/weapons/guns/fire/tank_cannon2.ogg', 35, TRUE),
+			playsound(src, 'sound/voice/pred_helpme.ogg', 35, TRUE))
+		if(2)
+			for(var/atom/movable/object AS in turf_contents)
+				if(isfood(object))
+					qdel(object)
+					playsound(src,'sound/items/eatfood.ogg', 25, 1)
+					balloon_alert_to_viewers("Consumes [object]")
+					break
+		if(3)
+			for(var/dirn in shuffle(GLOB.alldirs))
+				var/turf/destturf = get_step(src,dirn)
+				if(get_teleport_prereqs(destturf, TRUE))
+					src.forceMove(destturf)
+					break
+		if(4)
+			new /obj/item/tool/kitchen/knife/butcher(targetturf)
+			new /obj/effect/decal/cleanable/blood(targetturf)
+			color = COLOR_DARK_RED
+			desc += " It's covered in a dried reddish liquid, probably cranberry juice."
+		if(5)
+			pick(balloon_alert_to_viewers("stares"),
+			(balloon_alert_to_viewers("adjusts its hat")),
+			(balloon_alert_to_viewers("mimes a quick stabbing motion")),
+			(balloon_alert_to_viewers("rolls its eyes")),
+			(balloon_alert_to_viewers("mutters something")),
+			(balloon_alert_to_viewers("darts its eyes back and forth")),
+			(balloon_alert_to_viewers("stifles a laugh")),
+			(balloon_alert_to_viewers("blinks")),
+			(balloon_alert_to_viewers("squints")),
+			(balloon_alert_to_viewers("glares")),
+			(balloon_alert_to_viewers("[src]'s eyes gleam malevolently")))
+		if(6)
+			for(var/atom/movable/object AS in turf_contents)
+				if(isinjector(object))
+					qdel(object)
+					playsound(src,'sound/items/hypospray.ogg', 25, 1)
+					balloon_alert_to_viewers("Injects [object] into its arm")
+					break
+		if(7)
+			flick("gnome_hop", src)
+		if(8)
+			teleport_retries += 10 //gnome is getting out of here
+			teleport_routine()
+
+
+
+//handles gnome "escaping" a shuttle crush
+/obj/item/toy/plush/gnome/living/proc/shuttle_crush()
+	SIGNAL_HANDLER
+	new /obj/item/toy/plush/gnome(gnome_origin) //shuttle crush deletes src object, create new gnome at spawn point to give illusion of escape
+
+///handles gnome transportation using pipes
+/obj/item/toy/plush/gnome/living/proc/handle_pipe_mode(turf/targetturf)
+	pipe_mode_chance -= length(possible_mobs) * 10 //for each mob nearby subtract 10 from the chance to interact with vents
+	if(!prob(pipe_mode_chance))
+		return
+	if(pipe_mode)
+		var/list/pumplist = list()
+		for(var/obj/machinery/atmospherics/components/unary/vent_pump/targetpump in range(120, src))
+			pumplist += targetpump
+		if(length(pumplist))
```

Comment:
this is VERY LAGGY, honestly I'd just skip the vent stuff this code looks questionable at best